opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -67,26 +67,44 @@ impl Tracer {
     #[allow(clippy::too_many_arguments)]
     fn make_sampling_decision(
         &self,
-        parent_context: Option<&SpanContext>,
+        parent_cx: Option<&Context>,
+        parent_span_context: Option<&SpanContext>,
         trace_id: TraceId,
         name: &str,
         span_kind: &SpanKind,
         attributes: &[KeyValue],
         links: &[Link],
-        ctx: &Context,
     ) -> Option<(u8, Vec<KeyValue>, TraceState)> {
         let provider = self.provider()?;
         let sampler = &provider.config().default_sampler;
 
-        let ctx = parent_context.map(|span_context| {
-            let span = Span::new(span_context.clone(), None, self.clone());
-            ctx.with_span(span)
-        });
-
-        let sampling_result =
-            sampler.should_sample(ctx.as_ref(), trace_id, name, span_kind, attributes, links);
+        // Sampler needs to be able to access remote span context as parent span
+        let sampling_result = if parent_cx.map_or(false, |cx| {
+            !cx.has_active_span() && cx.remote_span_context().is_some()
+        }) {
+            let parent_cx = parent_cx.map(|cx| {
+                let span = Span::new(
+                    parent_span_context
```

Comment:
Maybe use `cx.remote_span_context()` would be more intuitive? I think if `!cx.has_active_span() && cx.remote_span_context().is_some()` is true. Parent span context should be equal to the remote span context in parent context.

Reply:
hm yeah the paths are a little mixed. Trying to not to duplicate the span context extraction, and not clone the context unnecessarily. Might be a clearer way.

Reply:
Simplified it a bit, now there is only one param for sampling